### DISCUSSION TITLE
How to set cache-control headers ?

### DISCUSSION THREAD

**Discussion by ghost** [vercel/vercel#7807] [Help] [answered]
Hello !
I have a frontity project (not next.js) and want to include stale-while-revalidate headers

However i have no idea where to add it ?

Doc says "Note: Please note that Cache-Control headers cannot be set using next.config.js or vercel.json (for other frameworks) files."
So vercel.json is not the way
https://vercel.com/docs/concepts/functions/serverless-functions/edge-caching

And FAQ mentions that vercel does not support .htaccess files
https://vercel.com/support/articles/does-vercel-support-htaccess-files

Hence, where do i add cache-control headers ??

Thank you ! :)

**Comment by swarnava** [comment, marked as answer]
Hi there, you should be able to create vercel.json in your root folder and set the header. Learn more here: https://vercel.com/docs/project-configuration

**Reply by ustun-ed** [reply]
seems like this is no longer working. See https://github.com/vercel/vercel/issues/7953#issuecomment-1584701250

**Reply by leerob** [reply]
Please see our updated documented on caching here: https://vercel.com/docs/concepts/edge-network/caching
